File: AndroidWithJava/ActivityLifeCycle/app/src/main/java/com/example/activitylifecycle/MainActivity.java
package com.example.activitylifecycle;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        System.out.println("OnCreate  method called......");
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Intent intent=new Intent(this,result.class);
        startActivity(intent);
    }



    protected void onStart() {
        System.out.println("onStart method called......");
        super.onStart();
    }

    protected void onResume() {
        System.out.println("onResume(); method called......");
        super.onResume();
    }

    protected void onPause() {
        System.out.println("onPause method called......");
        super.onPause();
    }

    protected void onStop() {
        System.out.println("onStop method called......");
        super.onStop();
    }

    protected void onRestart() {
        System.out.println("onRestart method called......");
        super.onRestart();
    }
    protected void onDestroy() {
        System.out.println("onDestroy method called......");
        super.onDestroy();
    }






}

File: AndroidWithKotlin/CameraPermision/app/src/main/java/com/example/camerapermision/MainActivity.kt
package com.example.camerapermision

import android.app.Activity
import android.content.Intent
import android.content.pm.PackageManager
import android.graphics.Bitmap
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.provider.MediaStore
import android.widget.Button
import android.widget.Toast
import androidx.appcompat.widget.AppCompatImageView
import androidx.core.app.ActivityCompat
import androidx.core.content.ContextCompat
import androidx.core.content.PackageManagerCompat

class MainActivity : AppCompatActivity() {
    var appCompatImageView: AppCompatImageView? = null

    companion object {
        private const val CAMERA_PERMISSION_CODE = 1
        private const val CAMERA_REQUEST_CODE = 2
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        var btn_camera: Button = findViewById(R.id.button)

        appCompatImageView = findViewById(R.id.appCompatImageView)

        btn_camera.setOnClickListener() {
            if (ContextCompat.checkSelfPermission(
                    this, android.Manifest.permission.CAMERA
                ) == PackageManager.PERMISSION_GRANTED
            ) {
                val intent = Intent(MediaStore.ACTION_IMAGE_CAPTURE)
                startActivityForResult(intent, CAMERA_REQUEST_CODE)
            } else {
                ActivityCompat.requestPermissions(
                    this, arrayOf(android.Manifest.permission.CAMERA),
                    CAMERA_PERMISSION_CODE

                )
            }

        }

    }

    override fun onRequestPermissionsResult(
        requestCode: Int,
        permissions: Array<out String>,
        grantResults: IntArray
    ) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)

        if (requestCode == CAMERA_PERMISSION_CODE) {
            if (grantResults.isEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
                val intent = Intent(MediaStore.ACTION_IMAGE_CAPTURE)
                startActivityForResult(intent, CAMERA_REQUEST_CODE)

            } else {
                Toast.makeText(this, "oops you denied the permission ", Toast.LENGTH_SHORT).show()
            }
        }
    }


    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (resultCode == Activity.RESULT_OK) {
            if (requestCode == CAMERA_REQUEST_CODE) {

                val thumNail: Bitmap = data!!.extras!!.get("data") as Bitmap
                appCompatImageView?.setImageBitmap(thumNail)
            }

        }
    }

}